### PR TITLE
Clean up code for PageBuilder preview

### PR DESCRIPTION
**Pull request by camillevilla**
### JIRA issue link
A follow up to discussion in #650 https://github.com/department-of-veterans-affairs/diffusion-marketplace/pull/650#discussion_r1279660032

## Description - what does this code do?
- Cleans up code for generating page previews in `/admin/pages/:id`. 
  - Moves information about page component fields to the page component model
  - Preserves method that renders a list of practice IDs as their practice name
  - Render thumbnail for image preview

## Testing done - how did you test it/steps on how can another person can test it 
1. Log in as an admin
2. Create a page
3. Add one of each component 
4. Save page and confirm that information you entered is rendered in the preview on `/admin/pages/:id/`

## Screenshots, Gifs, Videos from application (if applicable)

Screenshot 2023-08-17 at 4 01 06 PM
Screenshot 2023-08-17 at 4 00 59 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/models/page_publication_component.rb [hunk 1]
```diff
@@ -14,6 +14,14 @@ class PagePublicationComponent < ApplicationRecord
                  on: [:create, :update],
                  if: Proc.new { |component| component.url.present? && component.url.chars.first != '/' }
 
+  FORM_FIELDS = { # Fields and labels in .arb form
+    published_in: 'Published in',
+    published_on_month: 'Month',
+    published_on_day: 'Day',
+    published_on_year: 'Year',
+    authors: 'Authors'
+  }
```
Comment (PhilipDeFraties) [inline app/models/page_publication_component.rb:23]: just missing the `.freeze` here

File: app/models/page_subpage_hyperlink_component.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class PageSubpageHyperlinkComponent < ApplicationRecord
     has_one :page_component, as: :component, autosave: true
+
+    FORM_FIELDS = {
+        url: 'Subpage URL suffix',
+        title: 'Subpage title',
+        description: 'Subpage description',
+        card: 'Add card styling'
+    }
```
Comment (PhilipDeFraties) [inline app/models/page_subpage_hyperlink_component.rb:9]: `.freeze` missing

File: app/models/page_block_quote_component.rb [hunk 3]
```diff
@@ -4,6 +4,12 @@ class PageBlockQuoteComponent < ApplicationRecord
 
   before_save :strip_p_tags_from_text
 
+  FORM_FIELDS = { # Fields and labels in .arb form
+    citation: 'Body Text',
+    text: 'Citation Text'
+  }
```
Comment (PhilipDeFraties) [inline app/models/page_block_quote_component.rb:10]: `.freeze` missing

**Review by PhilipDeFraties — APPROVED** [review]
Other than a few `.freeze`s missing from the constants everything looks good!

File: app/models/page_subpage_hyperlink_component.rb (hunk 2 above)
Comment (camillevilla) [reply app/models/page_subpage_hyperlink_component.rb]: 👍

**Comment by camillevilla** [comment]
@PhilipDeFraties Good catch! I pushed up another commit with `.freeze` applied consistently.

**Pull request merged**